gocode_test.go

```diff
@@ -0,0 +1,144 @@
+package main_test
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+// use `go build . && go test .` to run this
```

Comment:
I've actually found a nicer way to handle such thing by compiling the project directly in tests. So I'll change it to use that approach rather than a comment.

Reply:
fixed